cmd/query/app/grpc_handler.go

```diff
@@ -177,3 +190,123 @@ func (g *GRPCHandler) GetDependencies(ctx context.Context, r *api_v2.GetDependen
 
 	return &api_v2.GetDependenciesResponse{Dependencies: dependencies}, nil
 }
+
+// GetLatencies is the gRPC handler to fetch latency metrics.
+func (g *GRPCHandler) GetLatencies(ctx context.Context, r *metrics.GetLatenciesRequest) (*metrics.GetMetricsResponse, error) {
```

Comment:
recently we've seen some tickets showing that apparently with some client libs it's possible to send a request where r==nil, which crashes the server.

Reply:
I see you already have this check in g.newBaseQueryParameters, but should call this first then

Reply:
yeah, at least the grpc clients generated in this project fail to marshal a nil message over the wire; but I trust in what you've seen from other client libs, so I've added a nil check for the request.

Note, the original nil check in `g.newBaseQueryParameters` was just for the `MetricsQueryBaseRequest`, which is a common attribute among all the request objects, but it doesn't check the parent request for nil.